packages/core/ui-settings/core-ui-settings-server-internal/src/clients/ui_settings_client_factory.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import type { UiSettingsServiceOptions } from '@kbn/core-ui-settings-server-internal';
+import { UiSettingsClient } from './ui_settings_client';
+import { UiSettingsGlobalClient } from './ui_settings_global_client';
+
+export class UiSettingsClientFactory {
```

Comment:
File needs unit tests coverage (assert that the correct client is returned given provided type, and that all options are properly propagated to the underlying construction call)